types/cookiejar/index.d.ts

```diff
@@ -0,0 +1,123 @@
+     * match (see http://en.wikipedia.org/wiki/HTTP_cookie#Domain_and_Pat).
+     * @param cookie string or a cookie to work on
+     * @param requestDomain string argument is used to default the domain if it is not explicit in the cookie string
+     * @param requestPath string argument is used to set the path if it is not explicit in a cookie String
+     */
+    constructor(cookie: string | Cookie, requestDomain?: string, requestPath?: string);
+
+    /**
+     * the set-cookie: string for this cookie
+     */
+    toString(): string;
+
+    /**
+     * the cookie: string for this cookie
+     */
+    toValueString(): string;
+
+    /**
+     * parses the string onto this cookie or a new one if called directly
+     * @param cookie string to be parsed into a Cookie
+     * @param requestDomain string definind the requesting domain
+     * @param requestPath string defining the requesting path
+     */
+    parse(cookie: string, requestDomain?: string, requestPath?: string): Cookie;
+
+    /**
+     * returns true if the access_info allows retrieval of this cookie
+     * @param accessInfo CookieAccessInfo
+     */
+    matches(accessInfo: CookieAccessInfo): boolean;
+
+    /**
+     * returns true if the cookies cannot exist in the same space
+     * (domain and path match)
+     * @param cookie Cookie
+     */
+    collidesWith(cookie: Cookie): boolean;
+}
+
+declare class CookieJar {
+    /**
+     * class to hold numerous cookies from multiple domains correctly
+     */
+    constructor();
+
+    /**
+     * modify (or add if not already-existing) a cookie to the jar
+     * @param cookie string | Cookie
+     * @param requestDomain string argument is used to default the domain if it is not explicit in the cookie string
+     * @param requestPath string argument is used to set the path if it is not explicit in a cookie String
+     */
+    setCookie(cookie: string | Cookie, requestDomain?: string, requestPath?: string): Cookie | false;
+
+    /**
+     * modify (or add if not already-existing) a large number of cookies to the
+     * jar
+     * @param cookie string or list of strings defining cookies
+     * @param requestDomain string argument is used to default the domain if it is not explicit in the cookie string
+     * @param requestPath string argument is used to set the path if it is not explicit in a cookie String
+     */
+    setCookies(cookie: string | ReadonlyArray<string>, requestDomain?: string, requestPath?: string): ReadonlyArray<Cookie> | false;
+
+    /**
+     * get a cookie with the name and access_info matching
+     * @param cookieName string to be parsed into a Cookie
+     * @param accessInfo CookieAccessInfo
+     */
+    getCookie(cookieName: string, accessInfo: CookieAccessInfo): Cookie;
+
+    /**
+     * grab all cookies matching this access_info
+     * @param accessInfo CookieAccessInfo
+     */
+    getCookies(accessInfo: CookieAccessInfo): ReadonlyArray<Cookie>;
+}
+
+export default CookieJar;
```

Comment:
At runtime this package does not have a default export. see:

```shell
c:\test\dt>npm install cookiejar
+ cookiejar@2.1.2
added 1 package from 1 contributor and audited 63 packages in 2.015s

c:\test\dt>node
> var c = require("cookiejar");
undefined
> c;
{ CookieAccessInfo: { [Function: CookieAccessInfo] All: {} },
  Cookie: [Function: Cookie],
  CookieJar: [Function: CookieJar] }
```
Please see https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default for mode details.